### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by tjadevries**

### CODE REVIEW

**Comment by tjadevries** [comment]
The reference chapter of the docs does not display right, this is due to this error in pdocs: https://github.com/timothycrosley/pdocs/issues/35
Let's just accept that for now.

File: .github/workflows/application_settings-buid_deploy_docs_release.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            git commit -m "add docs for new version $RELEASE_VERSION"
+            echo "Redirecting stable to newly released version " $RELEASE_VERSION
+            rm -rf stable
+            ln -s $RELEASE_VERSION stable
+            git add stable
+            git commit -m "redirect stable to new version $RELEASE_VERSION"
+            cd ..
+
+      - name: Push changes
+        uses: ad-m/github-push-action@master
```
Comment (JJdeVries) [inline .github/workflows/application_settings-buid_deploy_docs_release.yml:77]: same here with master.

Also what does this action actually do? I think you could just do a `git push` in the previous step right? Possibly we will have to set the read/write permission in the repo settings:

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#preventing-github-actions-from-creating-or-approving-pull-requests

File: .github/workflows/application_settings-buid_deploy_docs_release.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            git checkout gh-pages
+            mv site "$RELEASE_VERSION"
+            git add "$RELEASE_VERSION"/
+            git commit -m "add docs for new version $RELEASE_VERSION"
+            echo "Redirecting stable to newly released version " $RELEASE_VERSION
+            rm -rf stable
+            ln -s $RELEASE_VERSION stable
+            git add stable
+            git commit -m "redirect stable to new version $RELEASE_VERSION"
+            cd ..
```
Comment (JJdeVries) [inline .github/workflows/application_settings-buid_deploy_docs_release.yml:74]: There is an extra `cd ..` here that can be removed.

File: docs/3-Files.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  section in the `json` file
+
+The order of sections and/or fields in the file does not have to adhere to the order
+in which fields have been specified in the Container - or Section classes.
+
+Presence of sections and/or fields in the files that are not defined in the classes goes
+by silently.
+
+Fields or complete sections defined in the classes can be absent in the files as long as
+default values have been specified for all fields that have been left out.
```
Comment (JJdeVries) [inline docs/3-Files.md:65]: maybe add: `, otherwise a ValueError is raised`

Comment (JJdeVries) [reply docs/3-Files.md:65]: Consequently something I'm now thinking of. One annoyance I can have with libraries, is raising a single error every time.

Do we actually collect all them missing / wrong values and raise them as one error? Or will I have to start the application again for every error that I'm fixing?

File: docs/4-Using_parameters.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+- The instance of a parameter container is accessed via a class method `get()`;
+- The parameter value is then obtained by chaining with the section name and the
+  parameter name;
+- The first invocation of `get()` will create the container instance, try to read values
+  from file and store it for future access in a private module global (a dictionary with
+  the class id as key);
+- A parameter container should not be instantiated directly by client code (although it
+  is possible to do so, e.g. for testing purposes);
+- If needed, you can set the path for the parameter file before the first invocation of
+  `get()`.
```
Comment (JJdeVries) [inline docs/4-Using_parameters.md:15]: All this documentation is pretty good for thinking about edge cases.

We should probably have errors / at the very least printing if someone calls `set_filepath` after an instance was already created. We should probably either raise an error, or reload the configuration, although the last one might result in very weird behavior in an application!

File: docs/4-Using_parameters.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    # the next statement will create the settings
+    name_var: str = MyExampleSettings.get().basics.name  # name_var == "the name"
+    # the next statement just gets that same instance
+    totals_var: int = MyExampleSettings.get().basics.totals  # totals_var == 2
+
+    # if you have edited the settings file, you can reload it (which will create a new instance)
+    name_var = MyExampleSettings.get(reload=True).basics.name
+
+    # change settings parameters programmatically using update
+    MyExampleSettings.update({"basics": {"name": "new and shiny name", "totals": 222}})
```
Comment (JJdeVries) [inline docs/4-Using_parameters.md:73]: might be nice to show here that we actually only provide the changed values, so only provide `name` and assert that the `totals` has not changed.

File: docs/3-Files.md (hunk 3 above)
Comment (JJdeVries) [reply docs/3-Files.md:65]: ohh there is an entire section on validation. Maybe we should add a link here?
```
For more info on data validation [click here](./5-handling_deviations.md)
```

**Review by JJdeVries — CHANGES_REQUESTED** [review]

File: .github/workflows/application_settings-buid_deploy_docs.yml
[the remark this answers is not in the document]
Comment (tjadevries) [reply .github/workflows/application_settings-buid_deploy_docs.yml:1]: corrected

File: .github/workflows/application_settings-buid_deploy_docs_release.yml
[the remark this answers is not in the document]
Comment (tjadevries) [reply .github/workflows/application_settings-buid_deploy_docs_release.yml:1]: corrected

File: .github/workflows/application_settings-buid_deploy_docs_release.yml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
```
[the remark this answers is not in the document]
Comment (tjadevries) [reply .github/workflows/application_settings-buid_deploy_docs_release.yml:19]: changed to false

File: .github/workflows/application_settings-buid_deploy_docs_release.yml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        run: poetry install --no-interaction --with docs
+
+      - name: Build html
+        run:
+          poetry run portray as_html --overwrite -o site
+
+      - name: Setup git config
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
```
[the remark this answers is not in the document]
Comment (tjadevries) [reply .github/workflows/application_settings-buid_deploy_docs_release.yml:60]: changed

File: .github/workflows/application_settings-buid_deploy_docs_release.yml (hunk 1 above)
Comment (tjadevries) [reply .github/workflows/application_settings-buid_deploy_docs_release.yml:77]: have set v0.6.0
Shall we for now stay with the github action? It works fine...

File: docs/3-Files.md (hunk 3 above)
Comment (tjadevries) [reply docs/3-Files.md:65]: added your suggestion

File: docs/4-Using_parameters.md (hunk 5 above)
Comment (tjadevries) [reply docs/4-Using_parameters.md:73]: adapted

File: docs/4-Using_parameters.md (hunk 4 above)
Comment (tjadevries) [reply docs/4-Using_parameters.md:15]: Good idea, added a check and the option to do reload

File: .github/workflows/application_settings-buid_deploy_docs_release.yml (hunk 2 above)
Comment (tjadevries) [reply .github/workflows/application_settings-buid_deploy_docs_release.yml]: removed

**Review by JJdeVries — APPROVED** [review]

**Pull request merged**